### PR TITLE
Introduce new modal system with BuyModal (fix Banxa reload issue) 

### PR DESCRIPTION
**Pull request by mvaivre**
The goal of this PR is to create a version for the Banxa team to check out.

I'll follow up with another PR where I migrate all the modals to the new system + add the `memo` check you added to `shared-react`.

### CODE REVIEW

File: apps/mobile-wallet/src/features/modals/AppModals.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    <ModalsContainer pointerEvents={openedModals.length > 0 ? 'auto' : 'none'}>
+      {openedModals.map((modal) => {
+        switch (modal.params.name) {
+          case 'BuyModal':
+            return <BuyModal id={modal.id} key={modal.id} />
+        }
+      })}
+    </ModalsContainer>
+  )
+}
```
Comment (nop33) [inline apps/mobile-wallet/src/features/modals/AppModals.tsx:38]: Do we even want to render anything if `openedModals.length === 0`?

```ts
if (openedModals.length === 0) return null
```

File: apps/mobile-wallet/src/features/buy/BuyModal.tsx [hunk 2]
```diff
@@ -64,77 +66,86 @@ const BuyModal = (props: BuyModalProps) => {
...
-            />
-          </ModalContent>
-        )}
-      />
-    </Portal>
+    <BottomModal
+      title={t('Buy')}
+      maximisedContent
+      noPadding
+      onClose={onClose}
```
Comment (nop33) [inline apps/mobile-wallet/src/features/buy/BuyModal.tsx:102]: We could replace the `onClose` with `id` and move the `onClose` definition in the `BottomModal`.

File: apps/mobile-wallet/src/screens/Dashboard/DashboardScreen.tsx [hunk 3]
```diff
@@ -154,7 +154,7 @@ const DashboardScreen = ({ navigation, ...props }: ScreenProps) => {
             <Button onPress={handleSendPress} iconProps={{ name: 'send' }} variant="contrast" round flex short />
             <Button onPress={handleReceivePress} iconProps={{ name: 'download' }} variant="contrast" round flex short />
             <Button
-              onPress={() => setIsBuyModalOpen(true)}
+              onPress={() => dispatch(openModal({ name: 'BuyModal' }))}
```
Comment (nop33) [inline apps/mobile-wallet/src/screens/Dashboard/DashboardScreen.tsx:157]: Same nit comment as in the start

**Review by nop33 — COMMENTED** [review]
👏

File: apps/mobile-wallet/src/features/buy/BuyModal.tsx [hunk 4]
```diff
@@ -64,77 +66,86 @@ const BuyModal = (props: BuyModalProps) => {
...
     `&secondaryColor=${theme.global.complementary.slice(1)}`
 
-  const handleModalCloseWhenFinished = (e: WebViewNavigation) => {
+  useEffect(() => {
+    if (!currentUrl) {
+      setCurrentUrl(banxaInitialURL)
+    }
+  }, [defaultAddress, theme, currentUrl, banxaInitialURL])
+
+  if (!defaultAddress) return null
```
[the remark this answers is not in the document]
Comment (mvaivre) [reply apps/mobile-wallet/src/features/buy/BuyModal.tsx:84]: Rebase issue

File: apps/mobile-wallet/src/features/buy/BuyModal.tsx (hunk 2 above)
Comment (mvaivre) [reply apps/mobile-wallet/src/features/buy/BuyModal.tsx:100]: Is it ok if I do that in a follow up PR?

File: apps/mobile-wallet/src/features/modals/AppModals.tsx (hunk 1 above)
Comment (mvaivre) [reply apps/mobile-wallet/src/features/modals/AppModals.tsx]: Done

File: apps/mobile-wallet/src/features/buy/BuyModal.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (mvaivre) [reply apps/mobile-wallet/src/features/buy/BuyModal.tsx]: Done

File: apps/mobile-wallet/src/features/buy/BuyModal.tsx [hunk 5]
```diff
@@ -17,35 +17,37 @@ along with the library. If not, see <http://www.gnu.org/licenses/>.
...
-const BuyModal = (props: BuyModalProps) => {
+const BuyModal = memo(({ id }: ModalBaseProp) => {
   const { t } = useTranslation()
   const navigation = useNavigation<NavigationProp<InWalletTabsParamList>>()
   const theme = useTheme()
   const webViewRef = useRef<WebView>(null)
   const insets = useSafeAreaInsets()
   const defaultAddress = useAppSelector(selectDefaultAddress)
   const [isDisclaimerAccepted, setIsDisclaimerAccepted] = useState(false)
+  const [currentUrl, setCurrentUrl] = useState<string>('')
```
[the remark this answers is not in the document]
Comment (mvaivre) [reply apps/mobile-wallet/src/features/buy/BuyModal.tsx]: Done

File: apps/mobile-wallet/src/components/BalanceSummary.tsx [hunk 6]
```diff
@@ -63,47 +63,37 @@ const BalanceSummary = ({ dateLabel, style, ...props }: BalanceSummaryProps) =>
...
-      </BalanceSummaryContainer>
-      <BuyModal isOpen={isBuyModalOpen} onClose={() => setIsBuyModalOpen(false)} />
-    </>
+      </TextContainer>
+
+      {totalBalance === BigInt(0) && !isLoadingAlphBalances && addressesStatus === 'initialized' && (
+        <ReceiveFundsButtonContainer>
+          <Button onPress={handleReceivePress} iconProps={{ name: 'download' }} variant="highlight" short round flex />
+          <Button
+            onPress={() => dispatch(openModal({ name: 'BuyModal' }))}
```
[the remark this answers is not in the document]
Comment (mvaivre) [reply apps/mobile-wallet/src/components/BalanceSummary.tsx]: Done

File: apps/mobile-wallet/src/screens/Dashboard/DashboardScreen.tsx (hunk 3 above)
Comment (mvaivre) [reply apps/mobile-wallet/src/screens/Dashboard/DashboardScreen.tsx]: Fixed

File: apps/mobile-wallet/src/features/buy/BuyModal.tsx (hunk 2 above)
Comment (nop33) [reply apps/mobile-wallet/src/features/buy/BuyModal.tsx:100]: Sure, can you add a TODO in this PR?

**Review by nop33 — APPROVED** [review]

File: apps/mobile-wallet/src/features/buy/BuyModal.tsx (hunk 2 above)
Comment (mvaivre) [reply apps/mobile-wallet/src/features/buy/BuyModal.tsx:100]: I've added a note in my [follow up PR](https://github.com/alephium/alephium-frontend/pull/794) :)

**Pull request merged**